modify so gauge always written at starting time

rjleveque opened the pull request:
Previously if `min_time_increment > 0` was set for a gauge it wouldn't capture the gauge value until this long after the starting time value `t1` specified for the gauge.

Analogous to change in https://github.com/clawpack/geoclaw/pull/636.

mandli commented:
FWIW the modified code works in the 2D tests contained in #298.  Does this need to be added to 1D and 3D?

rjleveque commented:
I fixed the 3D code and also moved setting `last_time` into a pre-existing loop on gauges rather than in a separate loop (made that change also in https://github.com/clawpack/geoclaw/pull/636).

The 1D code does not allow specifying `min_time_increment`.

mandli commented:
I have tested this locally and it all works.  Merging in by the end of the day unless there is something else that needs to be done.

rjleveque commented:
Thanks @mandli, I think it is ready to merge.
